### PR TITLE
Upgrade HDF5 library version

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Comment by giordano** [comment]
Do we need to increase the version number, too?

**Comment by musm** [comment]
I guess I forgot to bump the previous version in the PR?

**Comment by musm** [comment]
I did a minor version bump, since we upgraded the library version and also upgraded to include `libhdf5_hl`

File: H/HDF5/build_tarballs.jl [hunk 1]
```diff
@@ -2,13 +2,13 @@ using BinaryBuilder
 
 # Collection of sources required to build HDF5
 name = "HDF5"
-version = v"1.10.5"
+version = v"1.11.0"
```
Comment (giordano) [inline H/HDF5/build_tarballs.jl]: Where does this number come from?

Comment (musm) [reply H/HDF5/build_tarballs.jl]: Sorry I don't understand what you mean?

**Comment by giordano** [comment]
JLL packages don't follow semantic versioning, we just use the same version as the upstream package.

**Comment by musm** [comment]
Ok well we have a weird situation in that case, since the linux and macos versions are not the same as the windows versions

**Comment by musm** [comment]
We should be fine to keep it as is then.

**Comment by musm** [comment]
v1.10.5 best reflects the library version

**Comment by musm** [comment]
Is it possible to make a release after this is merged? THanks

**Comment by giordano** [comment]
Each merged pull request triggers a new release.  I'm waiting for https://github.com/JuliaRegistries/General/pull/6259 to be merged in order to avoid conflicts

**Comment by musm** [comment]
Thanks.

Is there however an issue with the version number since I want my users to use build `+1` as a minimum and the package manager ignores build numbers for minimum compat versions?

**Pull request merged**